### PR TITLE
Removed additional '?' in WFS request

### LINKED ISSUE #2043: Query panel issues two identical requests on search.
Every search with the query panel issues two identical queries.
I think it is supposed to issue a "hits" request and a regular one.

It also generates wrong wfs parameters (typo?):
".../wfs??service=WFS&..."
Note the double question mark

### PR DESCRIPTION
**Pull request by allyoucanmap**
Added an url parser on query to remove additional '?'

### CODE REVIEW

File: web/client/epics/wfsquery.js [hunk 1]
```diff
@@ -122,8 +123,10 @@ const getWFSFilterData = (filterObj) => {
 
 const getWFSFeature = (searchUrl, filterObj) => {
     const data = getWFSFilterData(filterObj);
+    const urlParsedObj = Url.parse(searchUrl);
+    const parsedUrl = urlParsedObj.protocol + '//' + urlParsedObj.host + urlParsedObj.pathname;
```
Comment (offtherailz) [inline web/client/epics/wfsquery.js]: You should mix the query string parameters (service, outputFormat) with the ones coming from the original URL. So you can keep additional parameters in the URL.

File: web/client/epics/wfsquery.js [hunk 2]
```diff
@@ -122,8 +123,16 @@ const getWFSFilterData = (filterObj) => {
...
+
+    const additionalParams = isString(urlParsedObj.query) ? urlParsedObj.query.split('&').filter((q) => q.substring(0, 7) !== 'service' && q.substring(0, 12) !== 'outputFormat' && q !== '') : [];
+    const additionalParamsString = additionalParams.length > 0 ? additionalParams.reduce((a, b) => a + '&' + b, '') : '';
+    const params = '?service=WFS&outputFormat=json' + additionalParamsString;
+
     return Rx.Observable.defer( () =>
-        axios.post(searchUrl + '?service=WFS&outputFormat=json', data, {
+        axios.post(parsedUrl + params, data, {
```
Comment (mbarto) [inline web/client/epics/wfsquery.js]: How is this fixing the original issue of duplicate requests?

File: web/client/epics/wfsquery.js [hunk 3]
```diff
@@ -8,13 +8,15 @@
...
+const Url = require('url');
 const {changeSpatialAttribute, SELECT_VIEWPORT_SPATIAL_METHOD, updateGeometrySpatialField} = require('../actions/queryform');
 const {CHANGE_MAP_VIEW} = require('../actions/map');
 const {FEATURE_TYPE_SELECTED, QUERY, featureTypeLoaded, featureTypeError, querySearchResponse, queryError, featureClose} = require('../actions/wfsquery');
 const FilterUtils = require('../utils/FilterUtils');
 const assign = require('object-assign');
-const {isString} = require('lodash');
+const {isString, isObject} = require('lodash');
 const {TOGGLE_CONTROL, setControlProperty} = require('../actions/controls');
+const querystring = require('querystring');
```
Comment (offtherailz) [inline web/client/epics/wfsquery.js]: what's this? I think is missing in package.json

File: web/client/epics/wfsquery.js [hunk 4]
```diff
@@ -122,8 +124,16 @@ const getWFSFilterData = (filterObj) => {
 
 const getWFSFeature = (searchUrl, filterObj) => {
     const data = getWFSFilterData(filterObj);
+
+    const urlParsedObj = Url.parse(searchUrl, true);
+    const parsedUrl = urlParsedObj.protocol + '//' + urlParsedObj.host + urlParsedObj.pathname + '?';
+    let params = isObject(urlParsedObj.query) ? urlParsedObj.query : {};
+    params.service = 'WFS';
+    params.outputFormat = 'json';
+    const paramsString = querystring.stringify(params);
```
Comment (offtherailz) [inline web/client/epics/wfsquery.js]: You don't need other libraries. You can do all with [Url.format](https://www.npmjs.com/package/url#urlformaturlobj).
Create a new object from urlParsedObj modifying the query object and call Url.format with the new object.

**Pull request merged**